src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -145,6 +145,49 @@ def apply(self) -> List[LogicalPlan]:
 
         return resulting_plans
 
+    def _should_skip_optimization_for_session(
+        self,
+    ) -> Optional[SkipLargeQueryBreakdownCategory]:
+        """Method to check if the optimization should be skipped based on the session state."""
+        if self.session.get_current_database() is None:
```

Comment:
how about schema? when there no current schema, that won't work also

Reply:
added a check anyways.